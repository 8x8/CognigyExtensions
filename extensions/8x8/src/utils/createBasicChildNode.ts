import type { INodeDescriptor } from '@cognigy/extension-tools';
import { createNodeDescriptor } from '@cognigy/extension-tools';

export interface CreateBasicChildNodeParams {
  type: string
  defaultLabel: string
  parentType: string
}
const createBasicChildNode = ({ type, defaultLabel, parentType }: CreateBasicChildNodeParams): INodeDescriptor => createNodeDescriptor({
  type,
  defaultLabel,
  parentType,
  constraints: {
    editable: false,
    deletable: false,
    creatable: false,
    movable: false,
    placement: {
      predecessor: {
        whitelist: []
      }
    }
  },
  appearance: {
    color: '#61d188',
    textColor: 'white',
    variant: 'mini'
  }
});

export default createBasicChildNode;
