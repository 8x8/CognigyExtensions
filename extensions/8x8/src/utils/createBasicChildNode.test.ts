import createBasicChildNode from './createBasicChildNode';

describe('createBasicChildNode', () => {
  it('should create a basic child node', () => {
    const type = 'getCustomer';
    const defaultLabel = 'Get Customer';
    const parentType = 'getCustomer';

    const node = createBasicChildNode({ type, defaultLabel, parentType });

    expect(node).toMatchSnapshot();
  });
});
