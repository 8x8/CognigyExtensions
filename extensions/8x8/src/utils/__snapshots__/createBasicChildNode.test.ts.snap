// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createBasicChildNode should create a basic child node 1`] = `
{
  "_id": undefined,
  "appearance": {
    "color": "#61d188",
    "showIcon": true,
    "textColor": "white",
    "variant": "mini",
  },
  "behavior": {
    "entrypoint": false,
    "stopping": false,
  },
  "constraints": {
    "collapsable": false,
    "creatable": false,
    "deletable": false,
    "editable": false,
    "movable": false,
    "placement": {
      "children": {
        "whitelist": [],
      },
      "predecessor": {
        "whitelist": [],
      },
      "successor": {},
    },
  },
  "defaultLabel": "Get Customer",
  "dependencies": {
    "children": [],
  },
  "fields": [],
  "form": [],
  "function": null,
  "parentType": "getCustomer",
  "preview": undefined,
  "sections": [],
  "summary": "",
  "tags": [],
  "tokens": [],
  "type": "getCustomer",
}
`;
